File: src/main/java/phss/wsworkcars/repositories/CarroRepository.java
package phss.wsworkcars.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import phss.wsworkcars.models.Carro;

import java.util.List;
import java.util.Optional;

public interface CarroRepository extends JpaRepository<Carro, Long> {
    Optional<Carro> findByTimestampCadastro(long timestampCadastro);
    List<Carro> findByCor(String cor);
    List<Carro> findByCombustivel(String combustivel);
    List<Carro> findByNumPortas(int numPortas);
    List<Carro> findByAno(String ano);
    List<Carro> findByModeloId(long modeloId);
}

File: src/main/kotlin/phss/wsworkcars/repositories/CarroRepository.kt
package phss.wsworkcars.repositories

import org.springframework.data.jpa.repository.JpaRepository
import phss.wsworkcars.models.Carro

interface CarroRepository : JpaRepository<Carro, Long> {
    fun findByTimestampCadastro(timestampCadastro: Long): Carro?
    fun findByModeloId(modeloId: Long): List<Carro>
    fun findByAno(ano: String): List<Carro>
    fun findByCombustivel(combustivel: String): List<Carro>
    fun findByNumPortas(numPortas: Int): List<Carro>
    fun findByCor(cor: String): List<Carro>
}